### PR TITLE
Baton changes from TG along with an antag bola nerf

### PR DESCRIPTION
**Pull request by SomeguyManperson**
inormormadatio n: tgstation/tgstation#45377
basically: stunbatons have their stun delayed by a short period, and deal stamina damage, classic/telescopic  batons only deal stamina damage and knockdown. If someone has the "pump up" chem in their system, the stunbaton stun will be decreased by 90% while still dealing full stamina damage.
Additionally, bolas are being hit to compensate for this; the reinforced and cult bola cause immobility for their duration rather than a full paralyze stun.

:cl:  Akrilla
rscadd: Adds a new ghetto chem, Pump-Up, which can be made with epinephrine and coffee. Quick to metabolize, but massively lowers stun baton stun time. 2 epi : 5 coffee makes 5 units.
rscadd: Maintenance Pump-Up - a ghetto autoinjector containing the new chem, which is now part of the maintenance loot spawns.
tweak: Stun baton applies a delayed stun, with initial hit dealing stamina damage, slowdown and other effects. The stun will be applied 2 seconds after, but two hits put them in stamina crit instantly. There's a delay of two seconds before the baton can stun again.
tweak: Regular batons, e.g. telescopic, detective's, and contractor baton now deal a two second knockdown, with stamina damage. Two hits stuns them with a stamina crit.
tweak: reinforced and cult bola now cause immobility instead of a hard stun
tweak: spraycans also do a short knockdown instead of a long stun
/:cl:

### CODE REVIEW

**Review by Partheo — APPROVED** [review]
Sprites good.

Also, I like this change personally too.

**Comment by Jarod1200** [comment]
I like that this takes combat out of a one click win into something slightly more dynamic. This will force people to use slightly more than 1% of their brain.

**Comment by Alexandria404** [comment]
We have  few people playing sec, and those that do are often times not that good since they are newish.
Those that used to main sec stopped doing so, since it is such an uninviting task.
We under no circumstances can make sec less inviting.

This change especially with the other PRs in combination is a nerf to sec and a buff to the tide.

Exactly the opposite direction we need to go in.

**Comment by MegaEmpirical** [comment]
I'm willing to test merge but I'm not optimistic.

**Comment by DiamondSentinel** [comment]
Definitely better than before, although 2 attacks for a stamina crit seems a bit strong still. It still will lead to the wonder that is a Head/Detective easily winning a 3 or 4v1 because they have instawin items.

**Comment by SCortesBar** [comment]
This seems worth a try. I've seen similar work in other servers and it was perfectly fine. It also will force sec officers to use their bolas, too which they should learn to.

**Comment by Moja10** [comment]
Having tried the telebaton changes on /tg/ for a while now, they are completely awful. Getting someone out of the department is extremely difficult, he counters it with a free chem found in maint, and you are basically powerless if there's more than one person, on revs heads will basically have no defense. However the baton change is not as bad as people think. Batonning should be a little less brainless, and it's still quite easy to keep someone down with a baton, however mass bonking multiple with one baton is notoriously harder, whether that's a bad thing or not, dunno. I would prefer if it wasn't literally hopeless for heads of staff during rev rounds unless they have riot armor and a shotgun.

**Comment by asmcint** [comment]
I'm not 100% sure how I feel about this. On the one hand, I feel like moving away from stun combat is a good move overall for the game, but from the perspective of a semi-frequent sec player(when I can actually play), I'm very apprehensive about such significant changes to the stun baton. It's always been a very high risk, high reward tool, and cutting away reward while still maintaining the full risk is a tough pill to swallow. I'd be a lot more comfortable personally if there were other changes which mitigated the need to take that risk so often in the first place. A mild boost to how much charge a disabler can hold by default, maybe.

I dunno, there's a lot of mixed feelings going into this one for me. When sec has to deal with so much shit, it's really hard to imagine just letting go of their biggest advantage. Still, at the very least, if this PR is testmerged when I can actually boot into Windows, I'll be more than happy to give it a fair shake and see if/how my opinion changes.

**Comment by drewnerd** [comment]
"Pump-up" is feature creep. I would not add an unobtainable, overpowered chem, for a stunbaton nerf.

Chemistry is not intuitive, a majority of players will not have access to chem. In _theory_ you are trying to balance a stun nerf with an item anyone can make - but in practice, your average joe _never actually makes the chem,_ so what's the point?

/TG/ had a similar problem with Adrenals. An ability that basically gave antags max stamina at all times. In theory, this was to counter stuns, but having the zoomies was SO powerful, it was expected meta for a long time, nearly every traitor used it.

For veterans who can make the chem, it's overpowered. For the average player, it's ineffective at nerfing stunbatons.

**Comment by Spl99** [comment]
I mainly pay security, it can be hard sometimes to try and stun someone with a baton as it is. If you're against a robust person or if there's any kind of lag against you it can be a disaster. I feel like this change may not be productive for encouraging people to play security especially considering we can have a good few rounds without security officers. 
I feel like this change may cause more issues than it's worth and will discourage even more people from playing security

**Comment by TheLinuxOS** [comment]
I played with this change on TG and it made the game much more fun, you arent immediately screwed if someone batons you. Adds more interesting situations to combat such as batoning someone after you are batoned to keep you in the fight, or popping a stim after being hit.

**Comment by MegaEmpirical** [comment]
That's exactly what I expect a non-sec player to say.

**Comment by maxion12345** [comment]
On one hand:

No more funny "dead because clicked first"

on other hand:

"cant stop the greytider, he's injected epi and drank coffee and is now breaking windows at mach 5"

also:
literally every other stun chem needs effort, this is a chem tailored to defeat sec and it can be made by an assistant with an epi-pen and coffee. Thats only going to encourage additional greytiding.

I, for one, am split. I don't all like where combat is decided by who gets clicked with funny orange rod first, but I also don't think sec needs massive nerfs. Going to negative, because I think this is a wrong step in the right direction.

**Comment by Ratvar** [comment]
Stunbaton being so op is the reason antags have such op things as cult stunmagic. Without nerfing one, you cannot nerf another - and being slapped by stunarm or ebow rather bad for officers. Plus, this nerfs non-security killing an unrobust officer and getting baton, people being -promoted- into shitcurity by HoP and anyone with stunprods, which **helps** officers. That tg did not shit on antags does not mean alexcar will not.              
You can also cut chemical itself, which seems to be a graytide only weapon, modularize pr further.
To security players like MegaEmpirical - security are not majority of players, game can get better if yours gets worse, same way making xenobio worse leaves game a better place

**Comment by Jarod1200** [comment]
yeah the free "pumped up chem" in maint seems kinda OP. maybe make it do some toxin damage or something while its in your system so you cant just be 100u of funni bootleg stimulants man

**Comment by Darkstick** [comment]
I like having more ghetto chem options, looting epipens to fight sec sounds like fun

**Comment by ghost** [comment]
@MegaEmpirical The baton change is fine and i've taken to maining sec for a while on TG.

Telebaton change is shit.

**Comment by ghost** [comment]
@Jarod1200  @maxion12345 

> 15% chance to cause loss of breath, and 5% to cause some jitter. If overdosed has 20% chance of causing loss of breath and 4 stamina damage, 15% to cause 2 toxin, 5% to drop held items, 5% to jitter and can also make you drool. Can be found in maintenance (Results in 5 parts instead of 7)

It already has an od and the effects you want. It also does nothing for the stamina damage the baton inflicts-it's worthless, basically.

**Comment by Spl99** [comment]
I honestly think this is a bad direction for security to go in. Going back to my points about lag or a robust player against sec. Security will now need to rely on their disabler which is already a multiple hit weapon and is still affected by the above and the boola (which i quite like as a tool) but is also very susceptible to missing. So I don't see how this is beneficial to making the game more combat orientated, especially against a department that is already limited yet needs to rely on weapons

**Comment by ghost** [comment]
@Spl99 Try it before you shit on it. The baton is still quite powerful.

It's virtually impossible to fight back once you get batonned still-you have enough stam damage to slow you to a crawl and dizzied movement besides. You can effectively move one tile in a random direction after the baton, or hit back once, before the stun kicks in.

If anything, it's also a bit of a buff-adrenals and antistun chems like meth don't impact stam damage too much, so unlike before where meth made you immune to batons, the guy gets up fast but he's still considerably slow, allowing for a followup baton.

The stunbaton is still one of the best melee weapons in the game with this-it can still easily 1v1 any melee weapon. The delayed stun is not nearly as impactful as you imagine-someone batonned is still quite fucked. They're moving at a snail's pace, can't control where they move, and down in a second. The second hit shuts them down entirely.

What this actually means is that the baton is far worse at handling crowds, as you need to twice-baton multiple people, and that delay is costly. It's not that you can't handle a crowd with it-you can still 3v1-it's just harder to. Flashbangs, etc, are superior for this purpose, as they should be.

Security is not nearly as impacted by this as you believe. They're just as effective.

It also opens the door to weakening antag stuns like cult talismans in turn, so there's that.

**Comment by Marmio64** [comment]
make stunbaton still instantly stun people with 20 melee armor or less when you target the head/torso, mostly would allow security to deal with the average tider just as well, but still allowing opportunities to protect yourself. For reference, armor vests and helmet provide 30 melee protection, so they'd protect you from the instant stun.

yes i know its a port and you didn't make it, but still

neutral for now anyhow

**Comment by ghost** [comment]
@Marmio64 it still handles tiders just fine by default

**Comment by Partheo** [comment]
Disclaimer: I think the chem could go. /tg/ are hyper powergamers so of course everyone is going to have some on them, on Yogs it could be better.

Y'all are dumbasses if you think this makes security any worse. Whoever has the baton in a fight is still nearly guaranteed to win, and security has *other* tools they can use.

I find this is a fantastic rebalance over the current state where whoever gets lucky and picks up the batons wins the fight 100% of the time. There's nothing more retarded than seeing a naked, hyped-up assistant fly through space to valid the fully equipped nukeop- AND WINNING- because he has a stunbaton. They are hilariously powerful.

This is a nerf to stunbatons, not gutting security. If we want to move away from disarm-spam, who-clicks-first combat, this is a great step to take. Security has tons of other tools they can use to keep the upper hand, so arguing that this "makes their job so much harder" is frankly just not trying hard enough. Also you're acting as if the stun baton doesn't stun anymore. It still does. The two second delay is just a huge amount of agency that feels really good for whatever player is on the beating end of the stick.

Anyone starting their point with "I play security..." is failing to understand that yes, while this might make your game slightly harder, this change is not about you. It makes the overall game experience WAY better for every player. The mere existence of the stunbaton warps balance around it like @Ratvar mentioned.

**Comment by Partheo** [comment]
> I feel like this change may cause more issues than it's worth and will discourage even more people from playing security

Security is and almost always has been the most popular role. I'd be extremely surprised if this changes at all, even if you deleted the baton entirely. People are still gonna play sec.

**Comment by gaminada** [comment]
I've tried it, i dont like it, honestly would not want it implemented.

**Comment by SomeguyManperson** [comment]
reminder that the disarm rework allows you to get  a short stun on people who are knocked down, which classic batons do alongside stamina damage

**Comment by Alexandria404** [comment]
> 
> 
> > I feel like this change may cause more issues than it's worth and will discourage even more people from playing security
> 
> Security is and almost always has been the most popular role. I'd be extremely surprised if this changes at all, even if you deleted the baton entirely. People are still gonna play sec.

Screenshot taken when reading this comment, not some select time where there happened to be no sec.

unknown

really makes you wonder

**Comment by Partheo** [comment]
> really makes you wonder

This is probably the dumbest take in this whole thread.

- the pr isn't even merged yet, so the problem is unrelated

- people don't play security just for the instant stun on the baton 

- it's fucking 3 am

**Comment by SomeguyManperson** [comment]
I've tried playing security and the main drawback is needing to DEAL with people (both other officers and the people you're arresting) and how much pressure is on you following the rules/space law properly while the people you have to deal with don't care/have that pressure, not you don't have good enough weapons. 
if you want to make security more inviting to play, giving them OP toys isn't going to help. Stronger enforcement of the "don't fuck with security just for fun" rule section would.

**Comment by Tipy1802** [comment]
We already have a MASSIVE greytiding problem with people breaking in places non stop and not bothering to ask if they can come in since “it’s just faster to break in”. You can’t prevent people from breaking in since everyone has tools and gloves and somehow everyone is a master hacker. At least heads were able to get tiders out of the their departments but now they won’t be able to do even that.

For the love of god can we stop buffing tiders and powergamers

**Comment by SomeguyManperson** [comment]
> We already have a MASSIVE greytiding problem with people breaking in places non stop and not bothering to ask if they can come in since “it’s just faster to break in”. You can’t prevent people from breaking in since everyone has tools and gloves and somehow everyone is a master hacker. At least heads were able to get tiders out of the their departments but now they won’t be able to do even that.
> 
> For the love of god can we stop buffing tiders and powergamers



> reminder that the disarm rework allows you to get a short stun on people who are knocked down, which classic batons do alongside stamina damage

**Comment by TheLinuxOS** [comment]
Officers already have the most powerful tool in the world against greytiders that are constantly breaking into departments and messing with people... its called ahelp. If you are complaining about changing this solely because UNGA GREYTIDE POWERFUL then maybe you should consider ahelping it when greytiders are robusting you and stealing all your sec gear.

**Comment by MegaEmpirical** [comment]
Most greytide is treated as an IC issue so that doesn't hold any weight.

**Comment by Georgeur** [comment]
Rick Randumb here. This indirectly buffs lethal weapons, and ranged options, maybe good, maybe bad, expect me to use lethals way more. But. WHY IN THE FUCK IS THERE AN ANTI SEC DRUG AS A MAINT SPAWN, WHY THE FUCK IS THE EFFECT SO STRONG, WHY THE FUCK IS THIS SEEN AS ANYTHING OTHER THAN ENCOURAGING GREYTIDING. Nerf stun batons? Sure. Give literally any job the ability to make an anti sec drug, so long as they buy a coffee, fucking no. The effect is so strong too.

**Comment by ghost** [comment]
@Georgeur 

It's not strong, it's fucking useless.

It does nothing for stamina.  It's noob bait-completely impotent. 

@Tipy1802

This doesn't buff tiders, but it does open the door to nerfing stuns overall.

**Comment by JamieD1** [comment]
Tested on server with staff, seems fine.

**Comment by Ludox235** [comment]
@AdamElTablawy 
Why do you still comment despite being banned from the server?

Also yeah I've seen the "nerfed" stunbaton in action and it's still a very good. It doesn't really gut security or anything.

Basically: yes please.

**Comment by Tipy1802** [comment]
Wait, is the “classic” baton the detective one? Really nerfing the detective as if he needed nerfs when all he has is the forensic scanner which on top of not being very fun to use is accessible to any officer once researched and roundstart to hos. The detective was already super fucked during conversion game modes like gang since he can’t do much detective duty during those and it’s super vulnerable yet he is mindshielded, this only makes the detective more vulnerable and “ghost simulator 2020” during gangs

**Comment by ghost** [comment]
@Ludox235 why not it takes five seconds

**Comment by Marmio64** [comment]
can you also make mechs not stun with punches anymore, since that is a major stun thing that needs to be removed, perhaps instead make it a small knockdown, like a 1 second knockdown.

**Comment by maxion12345** [comment]
Mech stuns are intentional, but I believe they are supposed to be a durand-only feature. Maybe just make the person fly back or something, it's a fuckin mech you shouldn't be melee combating a gygax anyways

**Comment by JamieD1** [comment]
Will be merging this if there is no opposition

File: code/game/objects/items/melee/misc.dm [hunk 1]
```diff
@@ -174,13 +174,14 @@
 	var/cooldown_check = 0 // Used interally, you don't want to modify
 
 	var/cooldown = 40 // Default wait time until can stun again.
-	var/stun_time_carbon = 60 // How long we stun for - 6 seconds.
-	var/stun_time_silicon = 0.60 // Multiplier for stunning silicons; if enabled, is 60% of human stun time.
+	var/knockdown_time_carbon = (1.5 SECONDS) // Knockdown length for carbons.
```
Comment (alexkar598) [inline code/game/objects/items/melee/misc.dm]: This isnt a define
```suggestion
	var/knockdown_time_carbon = 1.5 SECONDS // Knockdown length for carbons.
```

File: code/game/objects/items/melee/misc.dm [hunk 2]
```diff
@@ -174,13 +174,14 @@
...
+	var/stun_time_silicon = (5 SECONDS) // If enabled, how long do we stun silicons.
```
Comment (alexkar598) [inline code/game/objects/items/melee/misc.dm]:
```suggestion
	var/stun_time_silicon = 5 SECONDS // If enabled, how long do we stun silicons.
```

File: code/game/objects/items/melee/misc.dm [hunk 3]
```diff
@@ -291,7 +287,8 @@
 				user.do_attack_animation(target)
 
 			playsound(get_turf(src), on_stun_sound, 75, 1, -1)
-			target.Paralyze(stun_time_carbon)
+			target.Knockdown(knockdown_time_carbon)
+			target.adjustStaminaLoss(stamina_damage)
```
Comment (alexkar598) [inline code/game/objects/items/melee/misc.dm:305]: Wont this make the head's telebaton be able to put people into stamcrit and cuff them without external help?

**Review by alexkar598 — CHANGES_REQUESTED** [review]

File: code/game/objects/items/stunbaton.dm [hunk 4]
```diff
@@ -12,7 +12,10 @@
 	attack_verb = list("beaten")
 	armor = list("melee" = 0, "bullet" = 0, "laser" = 0, "energy" = 0, "bomb" = 50, "bio" = 0, "rad" = 0, "fire" = 80, "acid" = 80)
 
-	var/stunforce = 140
+	var/cooldown_check = 0
+
+	var/cooldown = (2 SECONDS)
```
Comment (alexkar598) [inline code/game/objects/items/stunbaton.dm]:
```suggestion
	var/cooldown = 2 SECONDS
```

**Review by alexkar598 — DISMISSED** [review]

File: code/game/objects/items/melee/misc.dm (hunk 3 above)
Comment (SomeguyManperson) [reply code/game/objects/items/melee/misc.dm:305]: yes but currently the telebaton can already stun long enough to cuff people in one hit or to chainstun up to 2 people on its own

**Comment by Tipy1802** [comment]
After playing with the changes I can say that I hate them. First of all did security really need a nerf? Security is so frustrating to play as and this just makes it worse, we have a problem of security being often understaffed and this would only make it worse. 
 But the worse part has to be the police and telebaton changes. The detective, while not supposed to chase crimianls, is supposed to arrest crimianls if they see the crime happen in front of them or security is understaffed (which happens a lot), making it so he can't stun criminals long enough to cuff them means that he can't do anything to stop criminals.
 The most annoying part of being a Head of Staff has to be dealing with shitters breaking in non stop into your department and this change only makes that ten times worse. Chasing a shitter through medbay so he won't steal all the medecine or break into genetics and become a mega shitter while I am supposed to keep someone alive and when I finally hit the shitter with my batton he gets up 1 second later and the chase continues till I do it 2 or 3 more times. NO NO NO why on earth do we need to buff greytiders? They are allready such a masive pain in the ass and they shouldn't even exist. This changes were made for TG, a powergaming, greytiding LRP shithole, something we try NOT to be

**Comment by Tipy1802** [comment]
Also detective baton should be balanced like an officer's baton or else detectives will just ask for one

**Comment by maxion12345** [comment]
after playing with this for awhile, its great for preventing the "one click now your dead" kind of stun combat. However, the PUMPUP and overall lowered stun times makes playing security during the 15 assistant rounds we've been having complete fucking hell

**Comment by Tipy1802** [comment]
Actually after playing for a while I do like the sec baton changes but the tele and police baton changes still suck since heads should be able to get shitters out of their departments easy.
 I also don't like the maint pill that bypasses stun, antags already have adrenalin and why should greytiders be immune to security or heads of staff taking action against their fuckery?

**Review by JamieD1 — APPROVED** [review]

**Pull request merged**